### ISSUE TITLE
can't start container via api  

### ISSUE THREAD

**Issue by ringtail** [docker/classicswarm#1897]
docker deamon log 

```
INFO[11283] GET /containers/7c7b5f46cf19e94254949e9909be4e2d3031e43555f4ff4a84c98181f92ed441/json
INFO[11283] POST /v1.21/containers/7c7b5f46cf19e94254949e9909be4e2d3031e43555f4ff4a84c98181f92ed441/start
INFO[11284] GET /v1.21/images/json?all=1
2016/02/26 22:40:04 http: panic serving @: runtime error: index out of range
goroutine 439412 [running]:
net/http.func·011()
        /usr/local/go/src/net/http/server.go:1130 +0xbb
github.com/docker/docker/daemon.(*Container).buildSandboxOptions(0xc208d6d800, 0x7fed4ce035a8, 0xc208e77800, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container_unix.go:522 +0x29eb
github.com/docker/docker/daemon.(*Container).connectToNetwork(0xc208d6d800, 0xc20915cd80, 0x12, 0xc20936d101, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container_unix.go:1047 +0x71e
github.com/docker/docker/daemon.(*Container).allocateNetwork(0xc208d6d800, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container_unix.go:956 +0x574
github.com/docker/docker/daemon.(*Container).initializeNetworking(0xc208d6d800, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container_unix.go:1101 +0x4c6
github.com/docker/docker/daemon.(*Container).Start(0xc208d6d800, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/container.go:297 +0x236
github.com/docker/docker/daemon.(*Daemon).ContainerStart(0xc208662640, 0xc208b752e7, 0x40, 0x0, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/daemon/start.go:47 +0x24f
github.com/docker/docker/api/server/router/local.(*router).postContainersStart(0xc20871f380, 0x7fed4c55bef0, 0xc208b90570, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680, 0xc208b90450, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/container.go:192 +0x162
github.com/docker/docker/api/server/router/local.*router.(github.com/docker/docker/api/server/router/local.postContainersStart)·fm(0x7fed4c55bef0, 0xc208b90570, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680, 0xc208b90450, 0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/router/local/local.go:134 +0x7b
github.com/docker/docker/api/server.func·004(0x7fed4c55bef0, 0xc208b90570, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680, 0xc208b90450,
0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:87 +0x7c7
github.com/docker/docker/api/server.func·003(0x7fed5155ba50, 0xc20802af28, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680, 0xc208b90450,
0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:66 +0x10e
github.com/docker/docker/api/server.func·002(0x7fed5155ba50, 0xc20802af28, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680, 0xc208b90450,
0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:49 +0x47c
github.com/docker/docker/api/server.func·001(0x7fed5155ba50, 0xc20802af28, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680, 0xc208b90450,
0x0, 0x0)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/middleware.go:27 +0x1dc
github.com/docker/docker/api/server.func·007(0x7fed4c55beb8, 0xc2087a6140, 0xc209632680)
        /usr/src/docker/.gopath/src/github.com/docker/docker/api/server/server.go:156 +0x265
net/http.HandlerFunc.ServeHTTP(0xc2087078a0, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680)
        /usr/local/go/src/net/http/server.go:1265 +0x41
github.com/gorilla/mux.(*Router).ServeHTTP(0xc208649630, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680)
        /usr/src/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x297
net/http.serverHandler.ServeHTTP(0xc2088b59e0, 0x7fed4c55beb8, 0xc2087a6140, 0xc209632680)
        /usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc208993ea0)
        /usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:1751 +0x35e
```

swarm master log 

```
time="2016-02-26T14:10:20Z" level=info msg="Processing HTTP request" elapsedTime=1.249609ms method=GET uri="/images/json?all=0"
time="2016-02-26T14:10:20Z" level=info msg="Creating container  with config &{Hostname: Domainname: User: AttachStdin:false AttachStdout:false AttachStderr:false ExposedPorts:map[] Tty:false OpenStdin:false StdinOnce:false Env:[] Cmd:0xc820210240 Image:registry.aliyuncs.com/ringtail/jenkins-node-slave:0.1 Volumes:map[] WorkingDir: Entrypoint:<nil> NetworkDisabled:false MacAddress: OnBuild:[] Labels:map[com.github.kostyasha.yad.DockerCloud:杭州测试环境 com.github.kostyasha.yad.DockerSlave:6c4191d9-538c-4c68-9efa-af970adfcb5e] StopSignal: VolumeDriver: Memory:0 MemorySwap:0 CpuShares:0 Cpuset: PortSpecs:[22/tcp] HostConfig:{Binds:[] ContainerIDFile: LxcConf:[] Memory:0 MemoryReservation:0 MemorySwap:0 KernelMemory:0 CpuShares:0 CpuPeriod:0 CpusetCpus: CpusetMems: CpuQuota:0 BlkioWeight:0 OomKillDisable:false MemorySwappiness:0 Privileged:true PortBindings:map[22/tcp:[{HostIp: HostPort:} {HostIp:0.0.0.0 HostPort:}]] Links:[] PublishAllPorts:false Dns:[] DNSOptions:[] DnsSearch:[] ExtraHosts:[cs.console.aliyun.com 106.11.55.31] VolumesFrom:[] Devices:[] NetworkMode: IpcMode: PidMode: UTSMode: CapAdd:[] CapDrop:[] GroupAdd:[] RestartPolicy:{Name: MaximumRetryCount:0} SecurityOpt:[] ReadonlyRootfs:false Ulimits:[] LogConfig:{Type: Config:map[]} CgroupParent: ConsoleSize:[0 0] VolumeDriver:}}"
time="2016-02-26T14:11:29Z" level=info msg="Processing HTTP request" elapsedTime=1m9.57046537s method=POST uri="/containers/create"
time="2016-02-26T14:11:48Z" level=error msg="HTTP error: Post http://docker.c00579aa599d843df95ca155f1b9a1def-node1.c00579aa599d843df95ca155f1b9a1def.tunnel-cn-hangzhou.aliyun-inc.com:8080/v1.21/containers/9e5413d867aa9d01ba4ccea553f4fc68f990b0be87713f2b5fd0d81af08f5f71/start: EOF" status=500
time="2016-02-26T14:11:48Z" level=error msg="HTTP error: 500 Internal Server Error: Post http://docker.c00579aa599d843df95ca155f1b9a1def-node1.c00579aa599d843df95ca155f1b9a1def.tunnel-cn-hangzhou.aliyun-inc.com:8080/v1.21/containers/9e5413d867aa9d01ba4ccea553f4fc68f990b0be87713f2b5fd0d81af08f5f71/start: EOF\n" status=500
time="2016-02-26T14:11:48Z" level=info msg="Processing HTTP request" elapsedTime=18.561975005s method=POST uri="/containers/9e5413d867aa9d01ba4ccea553f4fc68f990b0be87713f2b5fd0d81af08f5f71/start"
time="2016-02-26T14:11:49Z" level=info msg="Processing HTTP request" elapsedTime=1.01276ms method=GET uri="/containers/json?all=0&size=0"
time="2016-02-26T14:11:50Z" level=info msg="Processing HTTP request" elapsedTime=1.275409ms method=GET uri="/containers/json?all=0&size=0"
time="2016-02-26T14:11:50Z" level=info msg="Processing HTTP request" elapsedTime=1.078711ms method=GET uri="/images/json?all=0"
time="2016-02-26T14:11:50Z" level=info msg="Creating container  with config &{Hostname: Domainname: User: AttachStdin:false AttachStdout:false AttachStderr:false ExposedPorts:map[] Tty:false OpenStdin:false StdinOnce:false Env:[] Cmd:0xc8202b1da0 Image:registry.aliyuncs.com/ringtail/jenkins-node-slave:0.1 Volumes:map[] WorkingDir: Entrypoint:<nil> NetworkDisabled:false MacAddress: OnBuild:[] Labels:map[com.github.kostyasha.yad.DockerCloud:杭州测试环境 com.github.kostyasha.yad.DockerSlave:6c4191d9-538c-4c68-9efa-af970adfcb5e] StopSignal: VolumeDriver: Memory:0 MemorySwap:0 CpuShares:0 Cpuset: PortSpecs:[22/tcp] HostConfig:{Binds:[] ContainerIDFile: LxcConf:[] Memory:0 MemoryReservation:0 MemorySwap:0 KernelMemory:0 CpuShares:0 CpuPeriod:0 CpusetCpus: CpusetMems: CpuQuota:0 BlkioWeight:0 OomKillDisable:false MemorySwappiness:0 Privileged:true PortBindings:map[22/tcp:[{HostIp: HostPort:} {HostIp:0.0.0.0 HostPort:}]] Links:[] PublishAllPorts:false Dns:[] DNSOptions:[] DnsSearch:[] ExtraHosts:[cs.console.aliyun.com 106.11.55.31] VolumesFrom:[] Devices:[] NetworkMode: IpcMode: PidMode: UTSMode: CapAdd:[] CapDrop:[] GroupAdd:[] RestartPolicy:{Name: MaximumRetryCount:0} SecurityOpt:[] ReadonlyRootfs:false Ulimits:[] LogConfig:{Type: Config:map[]} CgroupParent: ConsoleSize:[0 0] VolumeDriver:}}"
time="2016-02-26T14:12:22Z" level=info msg="Processing HTTP request" elapsedTime=31.889272438s method=POST uri="/containers/create"
time="2016-02-26T14:12:57Z" level=error msg="HTTP error: Post http://docker.c00579aa599d843df95ca155f1b9a1def-node2.c00579aa599d843df95ca155f1b9a1def.tunnel-cn-hangzhou.aliyun-inc.com:8080/v1.21/containers/0db747c2b0f5b61e98eff2de4e0e4b6fe54a9557487fc27be1a5056b7e8a1667/start: EOF" status=500
```

container status 

```
CONTAINER ID        IMAGE                                                   COMMAND                  CREATED              STATUS              PORTS               NAMES
6f1e985f6463        registry.aliyuncs.com/ringtail/jenkins-node-slave:0.1   "/usr/sbin/sshd -D -p"   About a minute ago   Created                                 fervent_mccarthy
```

**Comment by ringtail** [comment]
versions 
```
Docker version 1.9.0-rc4, build e6f5a3c
swarm version 0.4.0 (HEAD)
```

**Comment by dongluochen** [comment]
@ringtail There have been a lot of changes and bug fixes since `Docker version 1.9.0-rc4`. Can you upgrade your docker daemon to [Docker 1.10.2](https://github.com/docker/docker/releases/tag/v1.10.2) and retry?

**Comment by dongluochen** [comment]
I'd also suggest using [latest swarm build](https://github.com/docker/swarm/releases/tag/v1.1.3-rc1).
